<launch>
  <!-- Input -->
  <arg name="input_autoware_engage" default="/api/autoware/get/engage" />
  <arg name="input_control_mode" default="/vehicle/status/control_mode" />
  <arg name="input_route" default="/planning/mission_planning/route" />
  <arg name="input_odometry" default="/localization/kinematic_state" />

  <!-- Output -->
  <arg name="output_autoware_state" default="/autoware/state" />
  <arg name="output_autoware_engage" default="/autoware/engage" />

  <!-- Service -->
  <arg name="service_shutdown" default="/autoware/shutdown" />
  <arg name="service_reset_route" default="/autoware/reset_route" />

  <!-- Config -->
  <arg name="config_file" default="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.param.yaml" />

  <!-- Parameter -->
  <arg name="update_rate" default="10.0" />
  <arg name="th_arrived_distance_m" default="1.0" />
  <arg name="th_stopped_time_sec" default="1.0" />
  <arg name="th_stopped_velocity_mps" default="0.01" />
  <arg name="disengage_on_route" default="true" />
  <arg name="disengage_on_goal" default="true" />

  <node pkg="autoware_state_monitor" exec="autoware_state_monitor" name="autoware_state_monitor" output="screen">
    <remap from="input/autoware_engage" to="$(var input_autoware_engage)"/>
    <remap from="input/control_mode" to="$(var input_control_mode)"/>
    <remap from="input/route" to="$(var input_route)"/>
    <remap from="input/odometry" to="$(var input_odometry)"/>

    <remap from="output/autoware_state" to="$(var output_autoware_state)"/>
    <remap from="output/autoware_engage" to="$(var output_autoware_engage)"/>

    <remap from="service/shutdown" to="$(var service_shutdown)"/>
    <remap from="service/reset_route" to="$(var service_reset_route)"/>

    <param from="$(var config_file)" />

    <param name="update_rate" value="$(var update_rate)" />
    <param name="th_arrived_distance_m" value="$(var th_arrived_distance_m)" />
    <param name="th_stopped_time_sec" value="$(var th_stopped_time_sec)" />
    <param name="th_stopped_velocity_mps" value="$(var th_stopped_velocity_mps)" />
    <param name="disengage_on_route" value="$(var disengage_on_route)" />
    <param name="disengage_on_goal" value="$(var disengage_on_goal)" />
  </node>
</launch>
